test/integration-tests/IntegrationTests.MongoDB/MongoDbTests.cs

```diff
@@ -0,0 +1,88 @@
+#if NETCOREAPP
+using System.Collections.Generic;
+#endif
+
+using System.Linq;
+using IntegrationTests.Helpers;
+using OpenTelemetry.ClrProfiler.Managed.Tagging;
+using Xunit;
+using Xunit.Abstractions;
+
+#if NETFRAMEWORK
+using IntegrationTests.Helpers.Comptability;
+#endif
+
+namespace IntegrationTests.MongoDB
+{
+    [Collection(MongoDbCollection.Name)]
+    public class MongoDbTests : TestHelper
+    {
+        private readonly MongoDbFixture _mongoDb;
+
+        public MongoDbTests(ITestOutputHelper output, MongoDbFixture mongoDb)
+            : base("MongoDB", output)
+        {
+            SetServiceVersion("1.0.0");
+
+            _mongoDb = mongoDb;
+        }
+
+        [Theory]
+        [Trait("Category", "EndToEnd")]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void SubmitsTraces(bool enableCallTarget)
+        {
+            SetCallTargetSettings(enableCallTarget);
+
+            int agentPort = TcpPortProvider.GetOpenPort();
+
+            using (var agent = new MockZipkinCollector(Output, agentPort))
+            using (var processResult = RunSampleAndWaitForExit(agent.Port, arguments: $"--mongo-db {_mongoDb.Port}"))
+            {
+                Assert.True(processResult.ExitCode >= 0, $"Process exited with code {processResult.ExitCode} and exception: {processResult.StandardError}");
+
+                var spans = agent.WaitForSpans(3, 500);
+                Assert.True(spans.Count >= 3, $"Expecting at least 3 spans, only received {spans.Count}");
```

Comment:
Maybe you want to use [FluentAssertions](https://fluentassertions.com/introduction)?

Reply:
I'll pass this at the moment. Don't have good structure for FluentAssertions in my mind at the moment. I think it should be different, cause `Assert.*` is more eye-catching than `Should()`. So FluentAssertions should be more grouped together to separate normal code and assertions with naked eye.

Reply:
OK